examples/inverse/multi_dipole_model.py

```diff
@@ -0,0 +1,156 @@
+# full timecourses using MNE, assigning activity to both dipoles at the same
+# time while preventing leakage between the two. We use a very low ``lambda``
+# value to ensure both dipoles are fully used.
+
+fwd, _ = mne.make_forward_dipole([dip_left, dip_right], bem, info)
+
+# Apply MNE inverse
+inv = make_inverse_operator(info, fwd, cov, fixed=True, depth=0)
+stc_left = apply_inverse(evoked_left, inv, method='MNE', lambda2=1E-6)
+stc_right = apply_inverse(evoked_right, inv, method='MNE', lambda2=1E-6)
+
+# Plot the timecourses of the resulting source estimate
+fig, axes = plt.subplots(nrows=2, sharex=True, sharey=True)
+axes[0].plot(stc_left.times, stc_left.data.T)
+axes[0].set_title('Left auditory stimulation')
+axes[0].legend(['Dipole 1', 'Dipole 2'])
+axes[1].plot(stc_right.times, stc_right.data.T)
+axes[1].set_title('Right auditory stimulation')
+axes[1].set_xlabel('Time (s)')
+fig.supylabel('Dipole amplitude')
+
+# We can also fit the timecourses to single epochs. Here, we do it for each
+# experimental condition separately.
+stcs_left = apply_inverse_epochs(epochs['left'], inv, lambda2=1E-6,
+                                 method='MNE')
+stcs_right = apply_inverse_epochs(epochs['right'], inv, lambda2=1E-6,
+                                  method='MNE')
+
+###############################################################################
+# To summarize and visualize the single-epoch dipole amplitudes, we will create
+# a detailed plot of the mean amplitude of the dipoles during different
+# experimental conditions.
+
+# Summarize the single epoch timecourses by computing the mean amplitude from
+# 60-90ms.
+amplitudes_left = []
+amplitudes_right = []
+for stc in stcs_left:
+    amplitudes_left.append(stc.crop(0.06, 0.09).mean().data)
+for stc in stcs_right:
+    amplitudes_right.append(stc.crop(0.06, 0.09).mean().data)
+amplitudes = np.vstack([amplitudes_left, amplitudes_right])
+
+# Visualize the epoch-by-epoch dipole ampltudes in a detailed figure.
+n = len(amplitudes)
+n_left = len(amplitudes_left)
+mean_left = np.mean(amplitudes_left, axis=0)
+mean_right = np.mean(amplitudes_right, axis=0)
+
+fig = plt.figure(figsize=(8, 4))
+plt.scatter(np.arange(n), amplitudes[:, 0], label='Dipole 1')
+plt.scatter(np.arange(n), amplitudes[:, 1], label='Dipole 2')
+transition_point = n_left - 0.5
+plt.plot([0, transition_point], [mean_left[0], mean_left[0]], color='C0')
+plt.plot([0, transition_point], [mean_left[1], mean_left[1]], color='C1')
+plt.plot([transition_point, n], [mean_right[0], mean_right[0]], color='C0')
+plt.plot([transition_point, n], [mean_right[1], mean_right[1]], color='C1')
+plt.axvline(transition_point, color='black')
+plt.xlabel('Epochs')
+plt.ylabel('Dipole amplitude')
+plt.suptitle('Single epoch dipole amplitudes')
+fig.text(0.30, 0.9, 'Left auditory stimulation', ha='center')
+fig.text(0.70, 0.9, 'Right auditory stimulation', ha='center')
+plt.legend()
```

Comment:
boo hiss state-based interface :) `fig, ax = plt.subplots(figsize=(8, 4))` then `ax.*` is better practice nowadays

Reply:
changed this to use the OO interface. Am I part of the hip crew now?

Reply:
> Am I part of the hip crew now?

You already were, now thankfully we don't have to kick you out :)